tests/epyccel/modules/arrays.py

```diff
@@ -397,7 +397,141 @@ def array_real_4d_C_array_initialization(x, y, a):
     from numpy import array
     tmp      = array([x, y], dtype='float')
     a[:,:,:,:] = tmp[:,:,:,:]
+##==============================================================================
+## TEST NESTED ARRAYS INITIALIZATION WITH ORDER C
+##==============================================================================
 
+@types('real[:,:,:]','real[:,:]', 'real[:,:]', 'real[:,:,:,:]')
```

Comment:
This has been discussed multiple times. It is not incompatible with pure python if the import is correctly included `from pyccel.decorators import types`. It can be replaced by `@template` to some degree, but it gives more flexibility to limit the number of cases generated. E.g:
```
@types('float[:]','int[:]')
@types('float[:,:]','int[:,:]')
@types('float[:,:,:]','int[:,:,:]')
def f(a, b):
    pass
```
which describes 3 cases where arrays have the same number of dimensions. The equivalent with templates would lead to 9 cases including cases where arrays have different numbers of dimensions which could potentially be problematic:
```
@template('T', ['float[:]','float[:,:]','float[:,:,:]'])
@template('T', ['int[:]','int[:,:]','int[:,:,:]'])
def f(a, b):
    pass
```

Reply:
We would have significantly fewer examples if we went through and changed the syntax in these files as many new tests are created by copying from a nearby test

Reply:
I meant just the ones in this PR. We have to start from somewhere 😉

Reply:
> This has been discussed multiple times. It is not incompatible with pure python if the import is correctly included `from pyccel.decorators import types`. It can be replaced by `@template` to some degree, but it gives more flexibility to limit the number of cases generated. E.g:
> 
> ```
> @types('float[:]','int[:]')
> @types('float[:,:]','int[:,:]')
> @types('float[:,:,:]','int[:,:,:]')
> def f(a, b):
>     pass
> ```
> 
> which describes 3 cases where arrays have the same number of dimensions. The equivalent with templates would lead to 9 cases including cases where arrays have different numbers of dimensions which could potentially be problematic:
> 
> ```
> @template('T', ['float[:]','float[:,:]','float[:,:,:]'])
> @template('T', ['int[:]','int[:,:]','int[:,:,:]'])
> def f(a, b):
>     pass
> ```

True, there are some use cases for the `@types` annotation, but in most instances they could be replaced with either the `@elemental` annotation or the "partial templates", once we get to finally implement them.

Reply:
> I meant just the ones in this PR. We have to start from somewhere :wink:

There are still quite a lot of tests in this PR so it will take more time to correct just these ones than it would to write a script to correct them all. I will create a PR to remove unnecessary `@types`.